cmd/contour/serve.go

```diff
@@ -292,21 +292,21 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		log.WithField("context", "xDS"))
 
 	// snapshotHandler is used to produce new snapshots when the internal state changes for any xDS resource.
-	snapshotHandler := contour.NewSnapshotHandler(snapshotCache, resources, log.WithField("context", "snapshotHandler"))
+	snapshotHandler := xdscache.NewSnapshotHandler(snapshotCache, resources, log.WithField("context", "snapshotHandler"))
 
 	// register observer for endpoints updates.
-	endpointHandler.Observer = contour.ComposeObservers(snapshotHandler)
+	endpointHandler.Observer = xdscache.ComposeObservers(snapshotHandler)
 
 	dnsLookupFamily, err := ParseDNSLookupFamily(ctx.DNSLookupFamily)
 	if err != nil {
 		return fmt.Errorf("failed to configure configuration file parameter cluster.dns-lookup-family: %w", err)
 	}
 
 	// Build the core Kubernetes event handler.
-	eventHandler := &contour.EventHandler{
+	eventHandler := &xdscache.EventHandler{
```

Comment:
I'd probably leave the `EventHandler` type in the `internal/contour` package, not sure it makes sense under `xdscache`.